common/src/main/java/whocraft/tardis_refined/api/event/TardisEvents.java

```diff
@@ -0,0 +1,43 @@
+package whocraft.tardis_refined.api.event;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.LevelAccessor;
+import whocraft.tardis_refined.common.capability.TardisLevelOperator;
+import whocraft.tardis_refined.common.tardis.IExteriorShell;
+
+public class TardisEvents {
+
+    public static final Event<TakeOff> TAKE_OFF = new Event<>(TakeOff.class, listeners -> (tardisLevelOperator, level, pos) -> Event.result(listeners, takeOff -> takeOff.onTakeOff(tardisLevelOperator, level, pos)));
+
+    public static final Event<Land> LAND = new Event<>(Land.class, listeners -> ((tardisLevelOperator, level, pos) -> {
+        for (Land listener : listeners) {
+            listener.onLand(tardisLevelOperator, level, pos);
+        }
+    }));
+
+    public static final Event<TardisEntry> TARDIS_ENTRY_EVENT = new Event<>(TardisEntry.class, listeners -> (shell, player, externalPos, level, direction) -> {
+        for (TardisEntry listener : listeners) {
+            listener.onEnterTardis(shell, player, externalPos, level, direction);
+        }
+    });
+
+    @FunctionalInterface
+    public interface TardisEntry {
+        void onEnterTardis(IExteriorShell shell, Player player, BlockPos externalPos, Level level, Direction direction);
+    }
+
+    @FunctionalInterface
+    public interface TakeOff {
+        EventResult onTakeOff(TardisLevelOperator tardisLevelOperator, LevelAccessor level, BlockPos pos);
+
+    }
+
+    @FunctionalInterface
+    public interface Land {
+        void onLand(TardisLevelOperator tardisLevelOperator, LevelAccessor level, BlockPos pos);
```

Comment:
This event, being called "OnLand" presents itself as "Once the TARDIS has landed" but is fired when it begins to land, not after. It would be beneficial as someone calling this API to have the option to run off of two stages of these two events.

1. When the flight state has been started. (Starting takeoff, starting landing)
2. When the flight state has been ended. (The TARDIS entering the vortex, the TARDIS completed its landing.)

Reply:
This event was supposed to be when the Tardis has fully landed, you got a entry point for that?